hypersync-client/src/decode.rs

```diff
@@ -1,13 +1,21 @@
+use crate::simple_types::Log;
 use alloy_dyn_abi::{DecodedEvent, DynSolEvent, Specifier};
-use anyhow::{anyhow, Context, Result};
+use anyhow::{Context, Result};
 use hypersync_format::LogArgument;
+use std::collections::HashMap;
 
-use crate::simple_types::Log;
+#[derive(Debug, Hash, Eq, PartialEq)]
+struct EventKey {
+    topic0: Vec<u8>,
+    num_topics: usize,
+}
+
+type DecoderMap = HashMap<EventKey, DynSolEvent>;
```

Comment:
I thought a hashmap would be more performant (at least for larger abis)

Reply:
Vec will have better performance up to a very big number of elements (500, 1000)